terminal-ui.js

```diff
@@ -165,63 +150,47 @@ term.on('key', (name) => {
 			currentEdit = keysToFlags[name]
 			currentText = flags[keysToFlags[name]] ?? ''
 		} else if (name === 'ENTER') {
-			if (modeList[selectedMode] === 'keyframes' && flags['--keyframes'] === undefined)
-				return redrawStage2() || term(`\n\n${localizeString('keyframes_file_needed')}`)
+			if (modeList[selectedMode] === 'keyframes' && flags['--keyframes'] === undefined) return redrawStage2() || term(`\n\n${localizeString('keyframes_file_needed')}`)
 			stage = 3
 		}
 	} else if (stage === 3) {
 		if (name === 'ENTER') {
-			if (!fs.existsSync(filename))
-				return redrawStage3() || term(`\n\n${localizeString('file_not_found')}`)
+			if (!fs.existsSync(filename) && !isValidHttpUrl(filename)) return redrawStage3() || term(`\n\n${localizeString('file_not_found')}`)
 			stage = 4
 			editingText = false
-
-		}
-		else {
-			if (name === 'BACKSPACE')
-				filename = filename.substring(0, filename.length - 1)
+		} else {
+			if (name === 'BACKSPACE') filename = filename.substring(0, filename.length - 1)
 			else if (name.length > 2) {
 				/* ignore inputs like LEFT, TAB, etc */
-			}
-			else
-				filename += name
+			} else filename += name
 		}
 	} else if (stage === 4) {
 		if (name === 'ENTER') {
 			stage = 5
 
-			if (!flags['--bitrate'])
-				flags['--bitrate'] = '1M'
-			if (!flags['--thread'])
-				flags['--thread'] = 2
-			if (!flags['--tempo'])
-				flags['--tempo'] = 2
-			if (!flags['--angle'])
-				flags['--angle'] = 360
-			if (!flags['--compression'])
-				flags['--compression'] = 0
+			if (!flags['--bitrate']) flags['--bitrate'] = '1M'
+			if (!flags['--thread']) flags['--thread'] = 2
+			if (!flags['--tempo']) flags['--tempo'] = 2
+			if (!flags['--angle']) flags['--angle'] = 360
+			if (!flags['--compression']) flags['--compression'] = 0
 			if (!flags['--output'])
-				// not perfect, but works well enough
-				flags['--output'] = `${path.join(path.dirname(filename), getFileName(filename))}_${modeList[selectedMode]}.webm`
-
+				if (isValidHttpUrl(filename)) {
```

Comment:
can you put brackets around this? doing something like

    if (foo)
        if (bar) {}
        else {}

is obviously valid, but I think it looks a little bad